modules/features2d/test/ocl/test_feature2d.cpp

```diff
@@ -16,7 +17,7 @@ namespace ocl {
     "../stitching/a3.png", \
     "../stitching/s2.jpg")
 
-PARAM_TEST_CASE(Feature2DFixture, Ptr<Feature2D>, std::string)
```

Comment:
I got the idea, to make detector initialization lazy and move it from GTest cases generator inside the test body. I propose to add comment with the solution description. Without the memory consumption knowledge the test design looks very strange and ambiguous.